src/Saunter/AsyncApiOptions.cs

```diff
@@ -27,6 +29,36 @@ public class AsyncApiOptions
         /// </summary>
         public Func<Type, string> SchemaIdSelector { get; set; } = DefaultSchemaIdFactory.Generate;
 
+        /// <summary>
+        /// A function that specifies if the member name of the enum should be used instead of its value.
+        /// </summary>
+        public Func<Type, bool> UseEnumMemberName { get; set; } = type =>
+        {
+            var jsonConverterAttribute = type.GetCustomAttribute<JsonConverterAttribute>();
+            return jsonConverterAttribute?.ConverterType == typeof(JsonStringEnumConverter)
+                || jsonConverterAttribute?.ConverterType?.FullName == "System.Text.Json.Serialization.JsonStringEnumMemberConverter"
+                || jsonConverterAttribute?.ConverterType == typeof(EnumMemberConverter);
+        };
+
+        /// <summary>
+        /// A function that returns the enum member name.
+        /// </summary>
+        public Func<Type, Enum, string> EnumMemberNameSelector { get; set; } = (type, val) =>
+        {
+            var converterType = type.GetCustomAttribute<JsonConverterAttribute>().ConverterType;
```

Comment:
Let's just add a null check here -- it would technically be possible for someone to change the `UseEnumMemberName` func to not check the presence of the `JsonConverterAttribute` which would make this throw NullReferenceException when dereferencing the return value `null` from `GetCustomAttribute<JsonConverterAttribute>()`.

```csharp
var converterType = type.GetCustomAttribute<JsonConverterAttribute>()?.ConverterType;
```